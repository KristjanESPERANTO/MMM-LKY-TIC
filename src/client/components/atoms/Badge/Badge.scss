@import "../../../styles/common";

/** @define badge */

.mmm-lky-tic-wrapper {
  .badge {
    border: 1px solid $highlighted-border;
    border-radius: 4px;
    padding: $size-xs;
    color: $highlighted-text;
  }
}
