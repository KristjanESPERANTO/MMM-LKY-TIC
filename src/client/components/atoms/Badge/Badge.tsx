import './Badge.scss';

export interface BadgeProps {
  text: string;
};

const Badge: React.FC<BadgeProps> = (props: BadgeProps) => {
  return (
    <span className="badge">
      {props.text}
    </span>
  );
};

export default Badge;
