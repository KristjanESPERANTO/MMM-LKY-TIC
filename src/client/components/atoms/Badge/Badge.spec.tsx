import renderer from 'react-test-renderer';
import Badge, { BadgeProps } from './Badge';

describe('Badge atom component', () => {
  const defaultProps: BadgeProps = {
    text: 'text',
  };

  it('should render correctly', () => {
    // given-when
    const tree = renderer.create(<Badge {...defaultProps} />).toJSON();

    // then
    expect(tree).toMatchSnapshot();
  });
});
