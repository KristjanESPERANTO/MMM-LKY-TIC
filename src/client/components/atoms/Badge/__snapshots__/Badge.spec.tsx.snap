// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Badge atom component should render correctly 1`] = `
<span
  className="badge"
>
  text
</span>
`;
